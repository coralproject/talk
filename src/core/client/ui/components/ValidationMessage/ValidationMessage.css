.root {
  composes: alertMessage from "talk-ui/shared/typography.css";
  position: relative;
  display: inline-flex;
  justify-content: flex-start;
  align-items: center;
  padding: calc(0.5 * var(--spacing-unit)) var(--spacing-unit);
  box-sizing: border-box;
  border-radius: var(--round-corners);
  border-left-width: calc(0.5 * var(--spacing-unit));
  border-left-style: solid;
}

.colorError {
  background-color: var(--palette-error-light);
  border-color: var(--palette-error-darkest);
  color: var(--palette-common-white);
}

.fullWidth {
  display: flex;
  width: 100%;
}

.icon {
  margin-right: var(--spacing-unit);
}
