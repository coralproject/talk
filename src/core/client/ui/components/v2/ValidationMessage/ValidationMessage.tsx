import React, { FunctionComponent, ReactNode } from "react";

import Message from "../Message";
import MessageIcon from "../Message/MessageIcon";

export interface ValidationMessageProps {
  /**
   * The content of the component.
   */
  children: ReactNode;
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * If set renders a full width message
   */
  fullWidth?: boolean;
}

const ValidationMessage: FunctionComponent<ValidationMessageProps> = props => {
  const { className, fullWidth, children, ...rest } = props;

  return (
    <Message
      color="error"
      className={className}
      fullWidth={fullWidth}
      {...rest}
    >
      <MessageIcon>warning</MessageIcon>
      {children}
    </Message>
  );
};

ValidationMessage.defaultProps = {
  fullWidth: false,
};

export default ValidationMessage;
