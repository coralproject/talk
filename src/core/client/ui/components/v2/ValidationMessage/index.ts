export { default, ValidationMessageProps } from "./ValidationMessage";
