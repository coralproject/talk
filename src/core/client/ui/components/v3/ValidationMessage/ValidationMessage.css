.root {
  font-family: var(--v2-font-family-primary);

  font-weight: var(--v2-font-weight-primary-semi-bold);
  font-size: var(--v2-font-size-2);
  line-height: var(--v2-line-height-3);

  color: var(--v2-colors-red-500);
}

.icon {
  margin-right: var(--v2-spacing-1);
}
