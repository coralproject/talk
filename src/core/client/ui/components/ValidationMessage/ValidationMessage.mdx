---
name: ValidationMessage
menu: UI Kit
---

import { Playground } from 'docz'
import ValidationMessage from './ValidationMessage.tsx'
import Flex from '../Flex'

# ValidationMessage

## Basic Use
<Playground>
  <Flex itemGutter direction="column">
    <ValidationMessage>Invalid characters. Try again</ValidationMessage>
    <ValidationMessage fullWidth>Account with this email address already exists. Try another email. </ValidationMessage>
  </Flex>
</Playground>


