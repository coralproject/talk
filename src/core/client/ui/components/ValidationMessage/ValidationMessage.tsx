import cn from "classnames";
import React from "react";
import { ReactNode, StatelessComponent } from "react";
import { withStyles } from "talk-ui/hocs";
import Icon from "../Icon";
import * as styles from "./ValidationMessage.css";

export interface ValidationMessageProps {
  /**
   * The content of the component.
   */
  children: ReactNode;
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Override or extend the styles applied to the component.
   */
  classes: typeof styles;
  /*
  * If set renders a full width message
  */
  fullWidth?: boolean;
}

const ValidationMessage: StatelessComponent<ValidationMessageProps> = props => {
  const { className, classes, fullWidth, children, ...rest } = props;

  const rootClassName = cn(
    classes.root,
    classes.colorError,
    {
      [classes.fullWidth]: fullWidth,
    },
    className
  );

  return (
    <div className={rootClassName} {...rest}>
      <Icon size="sm" className={classes.icon}>
        warning
      </Icon>
      {children}
    </div>
  );
};

ValidationMessage.defaultProps = {
  fullWidth: false,
};

const enhanced = withStyles(styles)(ValidationMessage);
export default enhanced;
