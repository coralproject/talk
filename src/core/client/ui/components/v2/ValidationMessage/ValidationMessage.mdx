---
name: ValidationMessage
menu: UI Kit
---

import { Playground } from 'docz'
import ValidationMessage from './ValidationMessage'
import HorizontalGutter from '../HorizontalGutter'

# ValidationMessage

## Basic Use
<Playground>
  <HorizontalGutter>
    <ValidationMessage>Invalid characters. Try again</ValidationMessage>
    <ValidationMessage fullWidth>Account with this email address already exists. Try another email. </ValidationMessage>
  </HorizontalGutter>
</Playground>


