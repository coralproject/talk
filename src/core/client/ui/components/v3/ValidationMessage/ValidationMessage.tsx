import cn from "classnames";
import React, { FunctionComponent } from "react";

import { FieldMeta, hasError } from "coral-framework/lib/form";
import { Flex, Icon } from "coral-ui/components/v2";
import { withStyles } from "coral-ui/hocs";

import styles from "./ValidationMessage.css";

interface Props {
  variant?: "text" | "none";
  classes: typeof styles;
  className?: string;
  meta: FieldMeta;
}

const ValidationMessage: FunctionComponent<Props> = ({
  classes,
  className,
  meta,
}) => {
  const rootClassName = cn(classes.root, className);

  return hasError(meta) ? (
    <div className={rootClassName}>
      <Flex alignItems="center">
        <Icon size="sm" className={classes.icon}>
          error
        </Icon>
        {meta.error || meta.submitError}
      </Flex>
    </div>
  ) : null;
};

const enhanced = withStyles(styles)(ValidationMessage);

export default enhanced;
