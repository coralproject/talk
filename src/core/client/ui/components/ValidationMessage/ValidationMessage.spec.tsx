import React from "react";
import TestRenderer from "react-test-renderer";

import { PropTypesOf } from "talk-ui/types";

import ValidationMessage from "./ValidationMessage";

it("renders correctly", () => {
  const props: PropTypesOf<typeof ValidationMessage> = {
    className: "custom",
    children: "Hello World",
  };
  const renderer = TestRenderer.create(<ValidationMessage {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});
