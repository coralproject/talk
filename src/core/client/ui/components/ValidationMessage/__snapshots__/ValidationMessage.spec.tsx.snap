// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ValidationMessage-root ValidationMessage-colorError custom"
>
  <span
    aria-hidden="true"
    className="Icon-root ValidationMessage-icon Icon-sm"
  >
    warning
  </span>
  Hello World
</div>
`;
